pkg/render/node.go

```diff
@@ -696,10 +700,13 @@ func (c *nodeComponent) nodeVolumes() []v1.Volume {
 		},
 	}
 
-	if c.cr.CalicoNetwork != nil &&
-		c.cr.CalicoNetwork.LinuxDataplane != nil &&
-		*c.cr.CalicoNetwork.LinuxDataplane == operatorv1.LinuxDataplaneBPF {
-		volumes = append(volumes, v1.Volume{Name: "bpffs", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs/bpf", Type: &dirMustExist}}})
+	if c.bpfDataplane() {
+		volumes = append(volumes,
+			// Volume for the containing directory so that the init container can mount the child bpf directory if needed.
+			v1.Volume{Name: "sys-fs", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs", Type: &dirMustExist}}},
+			// Volume for the bpffs itself, used by the main node container.
+			v1.Volume{Name: "bpffs", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs/bpf", Type: &dirMustExist}}},
```

Comment:
Isn't the `dirMustExist` going to be a problem in the AKS case where it doesn't exist initially? Or does that only come into play when the main node container tries to use the volume?

Reply:
I was worried about that so I tested it and I was relived to discover that the init container gets to run before this would become a problem.  So, it all works as we'd hoped.